subsys/net/l2/ethernet/ethernet.c

```diff
@@ -177,6 +181,11 @@ static enum net_verdict ethernet_recv(struct net_if *iface,
 	struct net_linkaddr *lladdr;
 	sa_family_t family;
 
+#if defined(CONFIG_NET_DSA)
```

Comment:
let's use if (IS_ENABLED()) here (just make sure net_dsa_recv exists as a dummy inline if the config is not set)